File: 백준/Bronze/2798. 블랙잭/블랙잭.py
n, m = map(int, input().split())
arr = list(map(int, input().split()))
res = 0

for i in range(n):
    for j in range(i + 1, n):
        for k in range(j + 1, n):
            card = arr[i] + arr[j] + arr[k]
            if card > m:
                continue
            else:
                res = max(res, card)
print(res)

File: 백준/Bronze/2798. 블랙잭/블랙잭.cc
#include <iostream>
using namespace std;
 
int main(){
    int n, m;
    cin >> n >> m;
    int arr[n];
    for(int i = 0; i < n; i++){
        cin >> arr[i];
    }
 
    int card;
    int max = 0;
    for(int i = 0; i < n - 2; i++){
        for(int j = i + 1; j < n - 1; j++){
            for(int k = j + 1; k < n; k++){
                card = arr[i] + arr[j] + arr[k];
                if(card <= m && max < card) max = card;
            }
        }
    }
    cout << max;
    return 0;
}